File: internal/content/testdata/bench/go/same-tree.go
func is_same_tree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if p == nil || q == nil || p.Val != q.Val {
		return false
	}
	return is_same_tree(p.Left, q.Left) && is_same_tree(p.Right, q.Right)
}

File: internal/content/testdata/bench/java/same-tree.java
class Solution {
    public boolean is_same_tree(TreeNode p, TreeNode q) {
        if (p == null && q == null) return true;
        if (p == null || q == null || p.val != q.val) return false;
        return is_same_tree(p.left, q.left) && is_same_tree(p.right, q.right);
    }
}
